web3/datastructures.py

```diff
@@ -21,30 +30,34 @@
 # https://docs.python.org/3/reference/datamodel.html#object.__hash__
 
 
-class ReadableAttributeDict(Mapping):
+pretty = TypeVar("pretty", bound=Any)
+
+
+class ReadableAttributeDict(Mapping[Any, Any]):
     """
     The read attributes for the AttributeDict types
     """
 
-    def __init__(self, dictionary, *args, **kwargs):
+    def __init__(self, dictionary: Dict[Any, Any], *args: Any, **kwargs: Any) -> None:
         self.__dict__ = dict(dictionary)
         self.__dict__.update(dict(*args, **kwargs))
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: Any) -> Any:
         return self.__dict__[key]
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Any]:
         return iter(self.__dict__)
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self.__dict__)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.__class__.__name__ + "(%r)" % self.__dict__
 
-    def _repr_pretty_(self, builder, cycle):
+    def _repr_pretty_(self, builder: pretty, cycle: bool) -> None:
```

Comment:
Not sure I understand the need for this `TypeVar`.  Seems like `Any` would suffice (and probably doesn't matter to us at all since we don't call this API)